From 5dbc21ba1cf8f1e110b91275106ea4454f8ca231 Mon Sep 17 00:00:00 2001
From: ckie <git-525ff67@ckie.dev>
Date: Fri, 6 May 2022 22:59:10 +0300
Subject: [PATCH 1/2] feat(xkeyboard): normalize %name% to old convention

---
 src/modules/xkeyboard.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/modules/xkeyboard.cpp b/src/modules/xkeyboard.cpp
index 53920640..95087cf0 100644
--- a/src/modules/xkeyboard.cpp
+++ b/src/modules/xkeyboard.cpp
@@ -5,6 +5,8 @@
 #include "modules/meta/base.inl"
 #include "x11/atoms.hpp"
 #include "x11/connection.hpp"
+#include <iostream>
+#include <algorithm>
 
 POLYBAR_NS
 
@@ -111,7 +113,9 @@ namespace modules {
   void xkeyboard_module::update() {
     if (m_layout) {
       m_layout->reset_tokens();
-      m_layout->replace_token("%name%", m_keyboard->group_name(m_keyboard->current()));
+      auto group_name = m_keyboard->group_name(m_keyboard->current());
+      std::transform(group_name.begin(), group_name.end(), group_name.begin(), ::tolower);
+      m_layout->replace_token("%name%", group_name.substr(0, 2));
       m_layout->replace_token("%variant%", m_keyboard->variant_name(m_keyboard->current()));
 
       auto const current_layout = m_keyboard->layout_name(m_keyboard->current());
-- 
2.39.2

