From 9b889a94cf126ac8fdba1ee10d2bc39f0e37eda9 Mon Sep 17 00:00:00 2001
From: ckie <git-525ff67@ckie.dev>
Date: Thu, 30 Dec 2021 21:08:17 +0200
Subject: [PATCH 2/2] libexpr: add blackhole InternalType to printValue

---
 src/libexpr/eval.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libexpr/eval.cc b/src/libexpr/eval.cc
index 573bb1e33..c85f14463 100644
--- a/src/libexpr/eval.cc
+++ b/src/libexpr/eval.cc
@@ -157,6 +157,9 @@ void printValue(std::ostream & str, std::set<const Value *> & active, const Valu
     case tFloat:
         str << v.fpoint;
         break;
+    case tBlackhole:
+        str << "<BLACKHOLE>";
+        break;
     default:
         debug(format("invalid value %1%, aborting") % showType(v));
         abort();
-- 
2.34.1

