From 9e2eee888d99605a66e5da13f546df765054be48 Mon Sep 17 00:00:00 2001
From: ckie <git-525ff67@ckie.dev>
Date: Sat, 17 May 2025 11:31:13 +0300
Subject: [PATCH] feat(mpd): add titleOrFilename

---
 src/modules/mpd/mpd.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/modules/mpd/mpd.cpp b/src/modules/mpd/mpd.cpp
index 192e6c1a..063fc2f4 100644
--- a/src/modules/mpd/mpd.cpp
+++ b/src/modules/mpd/mpd.cpp
@@ -118,7 +118,7 @@ void waybar::modules::MPD::setLabel() {
 
   auto format = format_;
   Glib::ustring artist, album_artist, album, title;
-  std::string date, filename;
+  std::string date, filename, titleOrFilename;
   int song_pos = 0, queue_length = 0, volume = 0;
   std::chrono::seconds elapsedTime, totalTime;
 
@@ -151,6 +151,7 @@ void waybar::modules::MPD::setLabel() {
     title = sanitize_string(getTag(MPD_TAG_TITLE));
     date = sanitize_string(getTag(MPD_TAG_DATE));
     filename = sanitize_string(getFilename());
+    titleOrFilename = title == "N/A" ? filename : static_cast<std::string>(title);
     song_pos = mpd_status_get_song_pos(status_.get()) + 1;
     volume = mpd_status_get_volume(status_.get());
     if (volume < 0) {
@@ -184,7 +185,8 @@ void waybar::modules::MPD::setLabel() {
         fmt::arg("songPosition", song_pos), fmt::arg("queueLength", queue_length),
         fmt::arg("stateIcon", stateIcon), fmt::arg("consumeIcon", consumeIcon),
         fmt::arg("randomIcon", randomIcon), fmt::arg("repeatIcon", repeatIcon),
-        fmt::arg("singleIcon", singleIcon), fmt::arg("filename", filename));
+        fmt::arg("singleIcon", singleIcon), fmt::arg("filename", filename),
+        fmt::arg("titleOrFilename", titleOrFilename));
     if (text.empty()) {
       label_.hide();
     } else {
-- 
2.49.0

