From cd74630dbfc033850a24df68251249629eacaa2e Mon Sep 17 00:00:00 2001
From: ckie <git-525ff67@ckie.dev>
Date: Fri, 15 Oct 2021 18:35:38 +0300
Subject: [PATCH] terminfo: add 24-bit color support

---
 st.info | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/st.info b/st.info
index 8201ad6..4c01cfa 100644
--- a/st.info
+++ b/st.info
@@ -212,6 +212,17 @@ st-256color| simpleterm with 256 colors,
 	setab=\E[%?%p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m,
 	setaf=\E[%?%p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m,
 
+# Use colon separators.
+st-24bit|st with 24-bit direct color mode,
+  use=st-256color,
+  setb24=\E[48:2:%p1%{65536}%/%d:%p1%{256}%/%{255}%&%d:%p1%{255}%&%dm,
+  setf24=\E[38:2:%p1%{65536}%/%d:%p1%{256}%/%{255}%&%d:%p1%{255}%&%dm,
+# Use semicolon separators.
+st-24bits|st with 24-bit direct color mode,
+  use=st-256color,
+  setb24=\E[48;2;%p1%{65536}%/%d;%p1%{256}%/%{255}%&%d;%p1%{255}%&%dm,
+  setf24=\E[38;2;%p1%{65536}%/%d;%p1%{256}%/%{255}%&%d;%p1%{255}%&%dm,
+
 st-meta| simpleterm with meta key,
 	use=st,
 	km,
-- 
2.33.0

